From 11d4508de1f8753f30f88e3ec626d71bb59de697 Mon Sep 17 00:00:00 2001
From: Yannick LM <yannick@lorient.localdomain>
Date: Sat, 15 Aug 2009 18:48:33 +0200
Subject: [PATCH] Make CMake works on linux

  * We got to find acl_get_perm in acl/libacl.h
  * We got to link with acl libraray
---
 CMakeLists.txt |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b81f795..bf87386 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -603,6 +603,8 @@ IF(ENABLE_ACL)
   CHECK_LIBRARY_EXISTS(acl "acl_get_file" "" HAVE_ACL_LIB)
   IF(HAVE_ACL_LIB)
     SET(CMAKE_REQUIRED_LIBRARIES "acl")
+    FIND_LIBRARY(ACL_LIBRARY NAMES acl)
+    LIST(APPEND ADDITIONAL_LIBS ${ACL_LIBRARY})
   ENDIF(HAVE_ACL_LIB)
   #
   CHECK_FUNCTION_EXISTS_GLIBC(acl_create_entry HAVE_ACL_CREATE_ENTRY)
@@ -618,9 +620,9 @@ IF(ENABLE_ACL)
   # name, FreeBSD adds _np to mark it as "non-Posix extension."
   # Test for both as a double-check that we really have POSIX-style ACL support.
   CHECK_SYMBOL_EXISTS(acl_get_perm
-      "sys/types.h;sys/acl.h" HAVE_ACL_GET_PERM)
+      "sys/types.h;sys/acl.h;acl/libacl.h" HAVE_ACL_GET_PERM)
   CHECK_SYMBOL_EXISTS(acl_get_perm_np
-      "sys/types.h;sys/acl.h" HAVE_ACL_GET_PERM_NP)
+      "sys/types.h;sys/acl.h;acl/libacl.h" HAVE_ACL_GET_PERM)
   CHECK_SYMBOL_EXISTS(acl_get_link
       "sys/types.h;sys/acl.h" HAVE_ACL_GET_LINK)
   CHECK_SYMBOL_EXISTS(acl_get_link_np
-- 
1.6.4

